nativelib/src/main/scala/scala/scalanative/native/time.scala

```diff
@@ -1,44 +1,77 @@
 package scala.scalanative
 package native
 
-@extern
 object time {
 
   type time_t   = CLong
   type clock_t  = CLong
   type timespec = CStruct2[time_t, CLong]
   type tm       = CStruct9[CInt, CInt, CInt, CInt, CInt, CInt, CInt, CInt, CInt]
 
-  def asctime(time_ptr: Ptr[tm]): CString                               = extern
-  def asctime_s(buf: Ptr[CChar], bufsz: CSize, time_ptr: Ptr[tm]): CInt = extern
-  def clock(): clock_t                                                  = extern
-  def ctime(time: Ptr[time_t]): CString                                 = extern
-  def ctime_s(buffer: Ptr[CChar], bufsz: CSize, time: Ptr[time_t]): CInt =
-    extern
-  def difftime(time_end: CLong, time_beg: CLong): CDouble      = extern
-  def gmtime(time: Ptr[time_t]): Ptr[tm]                       = extern
-  def gmtime_s(time: Ptr[time_t], result: Ptr[tm]): Ptr[tm]    = extern
-  def localtime(time: Ptr[time_t]): Ptr[tm]                    = extern
-  def localtime_s(time: Ptr[time_t], result: Ptr[tm]): Ptr[tm] = extern
-  def mktime(time: Ptr[tm]): time_t                            = extern
+  private final val timeBufferSize = 26 // taken from `man asctime`
+  @inline private def zeroAlloc[T](size: CSize)(implicit z: Zone): Ptr[T] = {
+    val buf = z.alloc(size)
+    string.memset(buf, 0, size)
+    buf.cast[Ptr[T]]
+  }
```

Comment:
In bindings we try to stay 100% close to the original signatures and APIs in C. We do not introduce extra parameters or helper methods. It's not the goal of bindings to increase usability of C code, it's the goal of the javalib or 3-rd party projects to build higher-level abstractions.

Reply:
Ok, if that's strictly the aim of the project, that's fine. I was just put off by the functions that returned naked pointers to static buffers.